### PR TITLE
Lustre-aware mpi recipe

### PR DESCRIPTION
**Pull request by pelahi**
New recipe for lustre-aware mpi. Doing some further testing but initial tests are successful.

### CODE REVIEW

**Review by marcodelapierre — APPROVED** [review]
Tested build, could run and access container, Dockerfile looks good.

Only one request: can you post here what bind mounts are required (if any) in order to have a working combination of the container Lustre + Setonix Slurm ?

Thanks

**Comment by marcodelapierre** [comment]
Thanks for adding this info here, good for future reference

**Comment by pelahi** [comment]
@marcodelapierre @AlexisEspinosaGayosso , can you review. Having tested a parallel HDF5 read and write and also just MPI-IO I am happy with this recipe.

**Review by marcodelapierre — CHANGES_REQUESTED** [review]
This is amazing work, thank you @pelahi!

Successfully built the image, tested simple command outputs.

Please have a look at the few comments along the code.

File: mpi/lustrempich-base/buildlustrempich.dockerfile [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      && sed -i "s:CXX=CC:CXX=g++:g" ./build_cpu.sh \
+      && sed -i "s:MPICXX=CC:MPICXX=mpic++:g" ./build_cpu.sh \
+      && ./build_cpu.sh \
+      && cd examples/mpi/ \
+      && make MPICXX=mpic++ \
+      && cd ../../examples/openmp \
+      && make CXX=g++ bin/openmpvec_cpp
+
+# add mpi4py in the container 
+RUN pip install mpi4py
```
[the remark this answers is not in the document]
Comment (pelahi) [reply mpi/lustrempich-base/buildlustrempich.dockerfile]: same, now versioned

**Review by marcodelapierre — APPROVED** [review]

**Pull request merged**